libafl_frida/src/alloc.rs

```diff
@@ -205,6 +207,11 @@ impl Allocator {
         }
         let rounded_up_size = self.round_up_to_page(size) + 2 * self.page_size;
 
+        if self.total_allocation_size + rounded_up_size > self.options.asan_max_total_allocation() {
```

Comment:
Shouldn't we increment self.total_allocation_size only when we call mmap?
because if we successfully find a chunk from find_smallest_fit, we don't consume any additional memory in this alloc()

Reply:
no... we want to limit the total memory alloced per test, not the amount mmaped (subsequent tests will hardly ever mmap). I forgot to rest the total to zero when we reset the allocator.